[wip] Use more Linux shards

driazati opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#63641**
* #63640

This increases our test sharding to reduce total test runtime. Tests are all independent so we can parallelize them as much as we want to, it's just a question of how much overhead we have per job.

driazati committed:
Use more shards

[ghstack-poisoned]

janeyx99 reviewed (commented):
Yay! but is 20 the ideal amount/what's the plan to find a good balanced number of shards?

seemethere commented on .github/scripts/generate_ci_workflows.py:
```diff
@@ -247,7 +247,7 @@ def generate_workflow_file(self, workflow_template: jinja2.Template) -> None:
         on_pull_request=True,
         enable_doc_jobs=True,
-        num_test_shards=2,
+        num_test_shards=20,
```
20 is a bit unrealistic, we don't run true auto-scaling so this actually may make queue times longer since we do a 1:1 scaling for `check_run` events

driazati commented:
Yeah 20 is way too high, I was trying to test the sharding code to see if it could handle it and should have marked this with `[wip]`. Based on https://metrics.pytorch.org/d/L0r6ErGnk/github-status?orgId=1&from=now-24h&to=now&viewPanel=24 and spot checking a few jobs, it seems like going from 2 -> 6 shards for linux tests is good? spot checking a few jobs they are around 2 hours so this would cut it down to about 20 minutes each.

driazati committed:
Update on "[wip] Use more shards"




[ghstack-poisoned]

driazati committed:
Update on "[wip] Use more shards"




[ghstack-poisoned]

driazati committed:
Update on "[wip] Use more shards"




[ghstack-poisoned]

driazati committed:
Update on "[wip] Use more shards"




[ghstack-poisoned]

driazati committed:
Update on "[wip] Use more shards"




[ghstack-poisoned]

driazati committed:
Update on "[wip] Use more Linux shards"


This increases our test sharding to reduce total test runtime. Tests are all independent so we can parallelize them as much as we want to, it's just a question of how much overhead we have per job.


[ghstack-poisoned]

driazati committed:
Update on "[wip] Use more Linux shards"


This increases our test sharding to reduce total test runtime. Tests are all independent so we can parallelize them as much as we want to, it's just a question of how much overhead we have per job.


[ghstack-poisoned]

driazati committed:
Update on "[wip] Use more Linux shards"


This increases our test sharding to reduce total test runtime. Tests are all independent so we can parallelize them as much as we want to, it's just a question of how much overhead we have per job.


[ghstack-poisoned]
